src/Sql/Sql/Database/Services/AzureSqlDatabaseAdapter.cs

```diff
@@ -371,5 +373,19 @@ public static Sku GetDtuDatabaseSku(string requestedServiceObjectiveName, string
 
             return sku;
         }
+
+        public string GetServerlessSkuNameFromOtherParams(string edition, int vcore, string computeGeneration)
+        {
+            var shortEditions = new Dictionary<string, string>()
+            {
+                { "GeneralPurpose", "GP"},
+                { "BusinessCritical", "BC"},
+                { "Hyperscale", "HS"}
+            };
+            string shortEdition;
+            shortEditions.TryGetValue(edition, out shortEdition);
+
+            return string.Format(@"{0}_S_{1}_{2}", shortEdition, computeGeneration, vcore);
```

Comment:
Can't we just specify `"{0}_S"` ? Cmdlets should be already setting sku.family and sku.capacity, so they don't need to be repeated in the sku name.

Reply:
Here is example what returned from GetDatabase for Serverless:
sku: {name: "GP_S_Gen5", tier: "GeneralPurpose", family: "Gen5", capacity: 2}
capacity: 2
family: "Gen5"
name: "**GP_S_Gen5**"
tier: "GeneralPurpose"

So I can try to delete Capacity from name and check if it still works, but I am not sure if it will work without Gen in name.

Reply:
Understood, but that is the response. The input allows different combinations. Compare your code path against provisioned vcore, you will see that `GetDatabaseSkuName` above does not put in the family or capacity.

Reply:
Actually - I think you should just eliminate this method and change `GetDatabaseSkuName(string edition)` to `GetDatabaseSkuName(string tier, bool isServerless = false)`. Then push it down another layer, i.e. `SqlSkuUtils.GetVcoreSkuPrefix(string tier)` -> `SqlSkuUtils.GetVcoreSkuPrefix(string tier, bool isServerless = false)`. Then that method will return hardcoded "GP_S".